src/tests/system/tests/test_sssctl.py

```diff
@@ -288,424 +146,293 @@ def test_sssctl__check_ldap_host_object_class_in_domain(client: Client):
     ],
 )
-def test_sssctl__check_attribute_not_allowed_in_sssd(
-    client: Client, section: str, option: str, value: str, expected: str
-):
+@pytest.mark.tools
+@pytest.mark.topology(KnownTopology.Client)
+def test_sssctl__attribute_not_allowed_in_section(client: Client, section: str, option: str, value: str):
     """
-    :title: sssctl config-check validates attributes in specific sections
+    :title: sssctl config-check validates section attributes
     :setup:
-        1. Add an invalid option to a section in the configuration and start SSSD
+        1. Configure disallowed attributes in sections
     :steps:
-        1. Check the configuration using sssctl
+        1. Run sssctl config-check
     :expectedresults:
-        1. The config check succeed with the warning in the output
+        1. Configuration check fails
     :customerscenario: True
     """
     client.sssd.default_domain = "local"
-    client.sssd.common.local()
+    setup_local_sssd(client, start=False)
     getattr(client.sssd, section)[option] = value
     client.sssd.config_apply(check_config=False)
-
     result = client.sssctl.config_check()
-    assert result.rc != 0, "Config-check did not detect misconfigured config!"
-    assert expected in result.stdout, "Wrong error message on stdout!"
+    assert result.rc != 0, f"config-check should fail for {section}.{option}={value}"
 
 
-@pytest.mark.tools
 @pytest.mark.ticket(bz=1856861)
+@pytest.mark.tools
 @pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_enabling_2fa_prompting(client: Client):
+def test_sssctl__enabling_2fa_prompting(client: Client):
     """
-    :title: False warnings are logged in sssd.log file after enabling 2FA prompting
+    :title: sssctl config-check works with 2FA prompting
     :setup:
         1. Enable Two factor authentication
-        2. Start SSSD
     :steps:
-        1. Call sssctl config-check
+        1. Run sssctl config-check
     :expectedresults:
-        1. config-check succeed
+        1. Configuration check succeeds
     :customerscenario: True
     """
-    client.sssd.common.local()
+    setup_local_sssd(client, start=False)
     client.sssd.section("prompting/2fa/sshd")["first_prompt"] = "Enter OTP Token Value:"
     client.sssd.section("prompting/2fa/sshd")["single_prompt"] = "True"
-
     client.sssd.section("prompting/2fa")["first_prompt"] = "prompt1"
     client.sssd.section("prompting/2fa")["second_prompt"] = "prompt2"
     client.sssd.section("prompting/2fa")["single_prompt"] = "True"
-
     client.sssd.start(check_config=False)
     result = client.sssctl.config_check()
-    assert result.rc == 0, "Config-check failed"
+    assert result.rc == 0, "config-check failed with 2FA"
 
 
-@pytest.mark.tools
 @pytest.mark.ticket(bz=1791892)
+@pytest.mark.tools
 @pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_auto_private_groups_in_child_domains(client: Client):
+def test_sssctl__auto_private_groups_in_child_domains(client: Client):
```

Comment:
What is this test supposed to do? Rhetorical, it checks the configuration. We should have a good idea what the test does by the name alone.

```
test_sssctl_check_with_auto_private_groups_configured_with_subdomains
```